test/test.js

```diff
@@ -187,6 +187,71 @@ describe("S3rver Tests", function() {
     expect(data.ETag).to.match(/"[a-fA-F0-9]{32}"/);
   });
 
+  it("should store the text object directly to the bucket via a POST", function*() {
+    const key = "mytestimage.jpg";
+    const res = yield request({
+      method: "POST",
+      baseUrl: s3Client.config.endpoint,
+      url: `/${buckets[0]}`,
+      formData: {
+        key,
+        file: fs.createReadStream(path.join(__dirname, "resources/image0.jpg"))
+      },
+      resolveWithFullResponse: true
+    });
+    expect(res.statusCode).to.equal(204);
+    const data = yield s3Client
+      .getObject({ Bucket: buckets[0], Key: key })
+      .promise();
+    expect(data.ContentType).to.equal("binary/octet-stream");
```

Comment:
Can you add another expectation checking that the contents match? It would provide an important guarantee that the file is not corrupted during upload – no worries if it's too difficult, just reply in a comment.